packages/swing-store/src/transcriptStore.js

```diff
@@ -314,21 +329,124 @@ export function makeTranscriptStore(
     ORDER BY startPos
   `);
 
+  const sqlGetSomeVatSpans = db.prepare(`
+    SELECT vatID, startPos, endPos, isCurrent
+    FROM transcriptSpans
+    WHERE vatID = ?
+    ORDER BY startPos DESC
+    LIMIT ?
+  `);
+
+  const sqlDeleteVatSpan = db.prepare(`
+    DELETE FROM transcriptSpans
+    WHERE vatID = ? AND startPos = ?
+  `);
+
+  const sqlDeleteSomeItems = db.prepare(`
+    DELETE FROM transcriptItems
+    WHERE vatID = ? AND position >= ? AND position < ?
+  `);
+
+  /**
+   * Prepare for vat deletion by marking the isCurrent span as not
+   * current. Idempotent.
+   *
+   * @param {string} vatID  The vat being terminated/deleted.
+   */
+  function stopUsingTranscript(vatID) {
+    ensureTxn();
+    // this transforms the current span into a (short) historical one
+    const bounds = sqlGetCurrentSpanBounds.get(vatID);
+    if (bounds) {
+      const { startPos, endPos, hash, incarnation } = bounds;
+      sqlEndCurrentSpan.run(vatID);
+      // so we delete the transcript.${vatID}.current record, and add a
+      // .startPos one to replace it
+      noteExport(spanMetadataKey({ vatID, isCurrent: true }), undefined);
+      const newRec = spanRec(vatID, startPos, endPos, hash, false, incarnation);
+      noteExport(spanMetadataKey(newRec), JSON.stringify(newRec));
+    }
+  }
```

Comment:
This seems gratuitously divergent from `doSpanRollover`.
```suggestion
  function stopUsingTranscript(vatID) {
    ensureTxn();
    // this transforms the current span into a (short) historical one
    // (basically doSpanRollover without adding replacement data)
    const bounds = sqlGetCurrentSpanBounds.get(vatID);
    if (bounds) {
      // add a new record for the now-old span
      const { startPos, endPos, hash, incarnation } = bounds;
      const rec = spanRec(vatID, startPos, endPos, hash, false, incarnation);
      noteExport(spanMetadataKey(newRec), JSON.stringify(newRec));

      // and change its DB row to isCurrent=0
      sqlEndCurrentSpan.run(vatID);

      // remove the transcript.${vatID}.current record
      noteExport(spanMetadataKey({ vatID, isCurrent: true }), undefined);
    }
  }
```

Reply:
Yeah, I like your improvements, will apply